### DISCUSSION TITLE
Edge - evaluate 'Null' and 'Undefined' values as nothing

### DISCUSSION THREAD

**Discussion by Egorka** [adonisjs/core#2956] [Help]
Hello,

is there any option in Edge to evaluate 'Null' and 'Undefined' values in templates as nothing?

This is default behaviour in Handlebars and Nunjucks.

> Edge -> {{null}} -> null
> Handlebars -> {{null}} ->
> Nunjucks -> {{null}} ->

**Comment by WeyIin** [comment]
You can test variable in a model, if variable is null, then return empty string and use that computed property in the view.

e.g.
User.ts
```
@computed
public get variableToEmpty() {
  if(someVar === null) {
    return ''
 } else {
  return someVar
}
}
```

or you can test it in Edge with conditionals if you are expecting null

**Reply by Egorka** [reply]
Thank you @WeyIin,

as workaround I created nullToString method in model that convert all model null values to empty strings:

```
public nullToString() {
    return JSON.parse(JSON.stringify(this.serialize()).replace(/\:null/gi, "\:\"\""))
}
```

but I think it would be more logical to implement this feature in Edge core (possibly as an option) without extra steps as serialisation of model.

Since there is no reasons to display null and Undefined strings on website frontend.

**Comment by thetutlage** [comment]
I will explore the options. Does nunjucks or handlebars have a mode where they raise an exception too (especially with undefined)?

**Reply by Egorka** [reply]
Hello, @thetutlage,

as far as I know it's default behaviour in Handlebars and Nunjucks, without possibility to rise an exception on 'undefined' value.

[Nunjucks docs](https://mozilla.github.io/nunjucks/templating.html#variables):

> If a value is undefined or null, nothing is displayed. The same behavior occurs when referencing undefined or null objects. The following all output nothing if foo is undefined: {{ foo }}, {{ foo.bar }}, {{ foo.bar.baz }}.

**Reply by thetutlage** [reply]
I will not follow the nunjucks approach here. If `foo` is undefined then you cannot access `bar` and neither `bar.baz`. I will stick to the JavaScript behavior here
